nvme.c

```diff
@@ -2148,6 +2148,69 @@ static int get_ns_id(int argc, char **argv, struct command *cmd, struct plugin *
 	return 0;
 }
 
+static int virtual_mgmt(int argc, char **argv, struct command *cmd, struct plugin *plugin)
+{
+        const char *desc  = "The Virtualization Management command is supported by primary controllers "\
+                "that support the Virtualization Enhancements capability.This command is used for "\
+                "1. Modifying Flexible Resource allocation for the primary controller  "\
+                "2. Assigning Flexible Resources for secondary controllers; and "\
+                "3. Setting the Online and Offline state for secondary controllers";
+        const char *cntlid = "This field indicates the controller for which controller resources "\
+                "are to be modified.";
+        const char *rt = "Indicate the Resource Type (RT) : \n"\
+                "1. 0 - VQ Resources.\n"\
+                "2. 1 - VI Resources.\n\n";
+        const char *act = "Indicate the Action (ACT) : \n"\
+                "1. 1 - Primary Controller Flexible Allocation\n"\
+                "2. 7 - Secondary Controller Offline\n"\
+                "3. 8 - Secondary Controller Assign\n"\
+                "4. 9 - Secondary Controller Online";
+        const char *cdw11 = "Indicate the Number of Controller Resources (NR)";
+        int fd, err;
+ 
+        struct config {
+                int     cntlid;
+                int     rt;
+                int     act;
+                __u32   cdw10;
```

Comment:
cdw10 should just be normal stack variables rather than within the config struct. Variables in 'struct config' should only be there if they are command line parameters. For cdw11, you might want to give it a more descriptive name, like "nr".

If you want to provide cdw10 and 11 directly as an override, like for forward compatibility to access the currently reserved fields, that might be okay too.

Reply:
I agree and I will change n to nr, and what would you suggest for rest of them too like ctrlid for c, rtype for r(resource type) and act for a. I feel instead of letters if we have descriptive names easy to use and understand